lib/Cro/Iri/HTTP.pm6

```diff
@@ -0,0 +1,44 @@
+use Cro::Uri :decode-percents, :encode-percents;
+use Cro::HTTP::MultiValue;
+use Cro::Uri::HTTP;
+use Cro::Iri;
+use Cro::ResourceIdentifier::HTTP;
+
+class Cro::Iri::HTTP is Cro::Iri does Cro::ResourceIdentifier::HTTP {
+    grammar Parser is Cro::Iri::GenericParser {
+        proto token request-target { * }
+        token request-target:sym<origin-form> {
+            <iabsolute-path> [ "?" <iquery> ]?
+        }
+
+        token iabsolute-path {
+            [ "/" <isegment> ]+
+        }
+    }
+
+    grammar Actions is Cro::Iri::GenericActions {
+        method request-target:sym<origin-form>($/) {
+            make Cro::Iri::HTTP.bless(
+                path => $<iabsolute-path>.ast,
+                |(query => .ast with $<iquery>)
+            );
+        }
+
+        method iabsolute-path($/) {
+            make ~$/;
+        }
+    }
+
+    method parse-request-target(Str() $target) {
+        with Parser.parse($target, :actions(Actions), :rule('request-target')) {
+            .ast
+        }
+        else {
+            die X::Cro::Iri::ParseError.new(iri-string => $target)
+        }
+    }
```

Comment:
I don't think this makes sense, as a request is a URI by spec?

Reply:
What, trying to parse it and catch the exception? We still want to handle the failure to parse, no?